File: Linked list/remove_duplicates_from_sortedList.py
"""Given a sorted linked list, delete all duplicates 
such that each element appear only once."""

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
         
         
class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        if head is None:
            return
        curr = head
        while curr.next is not None:
            temp = curr.next
            if curr.val == curr.next.val:
                curr.next = temp.next
            else:
                curr = curr.next
        return head

File: Linked list/RemoveDuplicatesFromSortedList.cpp
//problem Link:- https://leetcode.com/problems/remove-duplicates-from-sorted-list/

#include<bits/stdc++.h>
using namespace std;

struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* curr = head;
        ListNode* tail = head;
        if (head == NULL){
            return head;
        }
        if (head->next == NULL){
            return head;
        }
        tail = head->next;
        while (curr != NULL && curr->next != NULL){
            while (tail != NULL && curr->val == tail->val){
                tail = tail->next;
            }
            curr->next = tail;
            curr = curr->next;
            if (tail != NULL){
                tail = tail->next;   
            }
        }
        return head;
    }
};